### PR TITLE
ONYX-26897 To reuse the token

### PR DESCRIPTION
**Pull request by Pooja-Gangwar**
### Desired Outcome

*Please describe the desired outcome for this PR.  Said another way, what was
the original request that resulted in these code changes?  Feel free to copy
this information from the connected issue.*

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue ID: [insert issue ID]

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: plugins/lookup/conjur_variable.py [hunk 1]
```diff
@@ -102,6 +102,7 @@
 from ansible.module_utils.urls import open_url
 from ansible.utils.display import Display
 import ssl
+from pathlib import Path
```
Comment (john-odonnell) [inline plugins/lookup/conjur_variable.py:105]: This import is not used in these changes

**Review by john-odonnell — COMMENTED** [review]
A few questions:
1. In the PR description, I notice a link to a Salesforce case, but I can't tell the connection between the case and this PR. Any more information here?
2. We definitely want a security review with @andytinkham before merging this to make sure we're correctly handling token files.
3. Caching the token will only improve performance on requests executed within the token's 8 minute lifetime. Could we implement a batch secret retrieval instead, without caching the token?
4. You've added some more test cases, but none of them actually confirm the changed behavior - we need at least one more test that confirms a retrieved Conjur token is cached in a particular file.

**Comment by Pooja-Gangwar** [comment]
> A few questions:
> 
> 1. In the PR description, I notice a link to a Salesforce case, but I can't tell the connection between the case and this PR. Any more information here?  Ans - Sorry I missed to mention here , I worked on 11/03/2022 Meeting's comments , now these comments are available in PR and in Jira ticket comments section these comments were already available . 
> 2. We definitely want a security review with @andytinkham before merging this to make sure we're correctly handling token files.  Ans - Sure, John
> 3. Caching the token will only improve performance on requests executed within the token's 8 minute lifetime. Could we implement a batch secret retrieval instead, without caching the token? 
> 4. You've added some more test cases, but none of them actually confirm the changed behavior - we need at least one more test that confirms a retrieved Conjur token is cached in a particular file. Ans -Sure

File: plugin_token.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+token
```
[the remark this answers is not in the document]
Comment (Pooja-Gangwar) [reply plugin_token.txt:1]: yes , it is there.

File: plugins/lookup/conjur_variable.py [hunk 3]
```diff
@@ -335,16 +345,29 @@ def run(self, terms, variables=None, **kwargs):
             display.vvv("Using cert file path {0}".format(conf['cert_file']))
             cert_file = conf['cert_file']
 
+        path = '../../tests/conjur_variable/plugin_token.txt'
+        isExist = os.path.exists(path)
```
[the remark this answers is not in the document]
Comment (Pooja-Gangwar) [reply plugins/lookup/conjur_variable.py:349]: Sure , I will update it .

[the remark this answers is not in the document]
Comment (Pooja-Gangwar) [reply plugins/lookup/conjur_variable.py]: Done

File: plugins/lookup/conjur_variable.py [hunk 4]
```diff
@@ -335,16 +345,29 @@ def run(self, terms, variables=None, **kwargs):
...
+                    identity['id'],
+                    identity['api_key'],
+                    validate_certs,
+                    cert_file
+                )
+                with open("plugin_token.txt", "wb") as binary_file:
+                    binary_file.write(token)
+            else:
+                with open("plugin_token.txt", "rb") as f:
+                    token = f.read()
```
[the remark this answers is not in the document]
Comment (Pooja-Gangwar) [reply plugins/lookup/conjur_variable.py]: Made changes to use the existing file "access_token" .

File: plugins/lookup/conjur_variable.py (hunk 1 above)
Comment (Pooja-Gangwar) [reply plugins/lookup/conjur_variable.py]: done

File: plugin_token.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Pooja-Gangwar) [reply plugin_token.txt:1]: now in new changes , I am not using any new file .

File: plugins/lookup/conjur_variable.py [hunk 5]
```diff
@@ -176,9 +177,18 @@ def _fetch_conjur_token(conjur_url, account, username, api_key, validate_certs,
...
+        return response.read()
+    if response.getcode() == 401:
+        raise AnsibleError('Conjur request has invalid authorization credentials as {0} and {1} response'.format(code, username))
+    if response.getcode() == 403:
+        raise AnsibleError('The controlling host\'s Conjur identity does not have authorization as \'{0}\' (got {1} response)'
                            .format(username, code))
+    if response.getcode() == 404:
+        raise AnsibleError('The token does not exist with {0} response '.format(code))
+    if response.getcode() == 500:
+        raise AnsibleError('Internal Server Error with {0} response'.format(code))
```
[the remark this answers is not in the document]
Comment (Pooja-Gangwar) [reply plugins/lookup/conjur_variable.py:190]: Shall I remove all except 200, 500 and 401 code . Please suggest .

**Comment by jtuttle** [comment]
Stale PR, closing.
